### PR TITLE
feat(tailscale): Tailscale Source plugin

### PR DESCRIPTION
**Pull request by candiduslynx**
Implements #5507

### CODE REVIEW

File: plugins/source/tailscale/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	grep -rl '// Code generated by codegen; DO NOT EDIT.' resources/services/* | xargs rm
+	go run codegen/main.go
+
+.PHONY: gen-docs
+gen-docs:
+	rm -rf ./docs/tables/*
+	go run main.go doc ./docs/tables
+#	sed 's_(\(.*\))_(https://github.com/cloudquery/cloudquery/blob/main/plugins/source/tailscale/docs/tables/\1)_' docs/tables/README.md > ../../../website/pages/docs/plugins/sources/tailscale/tables.md
+#	go run main.go doc --format json docs
+#	rm docs/__tables.json
```
Comment (hermanschaaf) [inline plugins/source/tailscale/Makefile:32]: These two lines are only necessary for plugins with policies, so we can safely delete them (the policy docs use the JSON output of tables)

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
Looks great! I left a few minor comments, and I think we still need to follow some of the steps in this document for it to be released: https://github.com/cloudquery/cloudquery/blob/main/contributing/adding_a_new_plugin_to_cq_monorepo.md

File: plugins/source/tailscale/main.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package main
+
+import (
+	"github.com/cloudquery/cloudquery/plugins/source/tailscale/resources/plugin"
+	"github.com/cloudquery/plugin-sdk/serve"
+)
+
+// TODO: fill in proper DSN
+const sentryDSN = "https://995c68a7e67541338e22dd8120e81c42@o1396617.ingest.sentry.io/4504316028452864"
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply plugins/source/tailscale/main.go:9]: No, the committed one is right

[the remark this answers is not in the document]
Comment (hermanschaaf) [reply plugins/source/tailscale/main.go:9]: Ok, I see - please fix the comment then though :)

File: website/versions/source-tailscale.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{ "latest": "plugins-source-tailscale-v1.0.0" }
```
Comment (bbernays) [inline website/versions/source-tailscale.json:1]: Shouldn't this be a pre `1.0.0` release as we should release it with a preview/alpha tag for feedback?

Comment (erezrokah) [reply website/versions/source-tailscale.json:1]: I think we keep all the versions in the same format e.g. `v1.0.0` without any `pre` for simplicity.
We only difference is how we document the plugins in https://www.cloudquery.io/docs/plugins/sources/overview

File: plugins/source/tailscale/client/spec.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	"github.com/tailscale/tailscale-client-go/tailscale"
+)
+
+type Spec struct {
+	// APIKey can be set via spec or via "TAILSCALE_API_KEY" environment variable
+	APIKey string `json:"api_key,omitempty"`
+
+	// Tailnet can be set via spec or via "TAILSCALE_TAILNET" environment variable
+	Tailnet string `json:"tailnet,omitempty"`
+
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply plugins/source/tailscale/client/spec.go:16]: done

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Nice! Looks good and I like the tests with http instead of mocking. 

Few minor comments:
- Makefile: remove `install-tools` and `pre-commit` as it's deprecated
- Why `interface.go` is needed? I don't think you need it and can use the specific tailscale.Client type.

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**